backend/beyond_the_loop/routers/payments.py

```
@@ -0,0 +1,420 @@
+            customer=stripe_customer_id,
+            auto_advance=True  # Automatically finalize the invoice
+        )
+
+        return invoice
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.post("/create-checkout-session/")
+async def create_checkout_session(user=Depends(get_verified_user)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)